stdlib/websub/src/main/ballerina/websub/h2_hub_persistence.bal

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) 2018 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/h2;
+import ballerina/log;
+
+const string CREATE_TOPICS_TABLE = "CREATE TABLE IF NOT EXISTS topics (topic VARCHAR(255), PRIMARY KEY (topic))";
+const string INSERT_INTO_TOPICS = "INSERT INTO topics (topic) VALUES (?)";
+const string DELETE_FROM_TOPICS = "DELETE FROM topics WHERE topic=?";
+const string SELECT_ALL_FROM_TOPICS = "SELECT * FROM topics";
+
+const string CREATE_SUBSCRIPTIONS_TABLE = "CREATE TABLE IF NOT EXISTS subscriptions
+                                            (topic VARCHAR(255), callback VARCHAR(255), secret VARCHAR(255),
+                                            lease_seconds BIGINT, created_at BIGINT, PRIMARY KEY (topic, callback))";
+const string INSERT_INTO_SUBSCRIPTIONS_TABLE = "INSERT INTO subscriptions
+                                                (topic,callback,secret,lease_seconds,created_at) VALUES (?,?,?,?,?)";
+const string DELETE_FROM_SUBSCRIPTIONS = "DELETE FROM subscriptions WHERE topic=? AND callback=?";
+const string SELECT_FROM_SUBSCRIPTIONS = "SELECT topic, callback, secret, lease_seconds, created_at FROM subscriptions";
+
+# Represents H2 based hub persistence configuration and functions.
+public type H2HubPersistenceStore object {
```

Comment:
AFAIU if we need to support for another db we need to rewrite all the logic again in H2HubPersistenceStore. It seems for new database only change will be on sql queries and client. Since we passing the client , is it feasible to determine the db type from the client . If it can be done, we just need to load only db specific queries ( select, delete and insert will be most of the case same to different database types) in __init method.

So we do not need specific HubPersistenceStore for each db type.

Reply:
The idea behind introducing an H2 based implementation, was to provide a default implementation for users who wanted to persist data, but did not want to implement their own store.

However, agreed that we could support other db types similarly too. Will open up an issue and introduce this separately, since this would require some changes and extensive testing. WDYT?

Reply:
+1

Reply:
Created https://github.com/ballerina-platform/ballerina-lang/issues/13183.